changelog.d/20231204_155243_chris_remove_funcx_client.rst

```diff
@@ -0,0 +1,4 @@
+Deprecated
+^^^^^^^^^^
+
+- The ``funcx_client`` argument to the ``Executor`` has been deprecated and replaced with ``compute_client``.
```

Comment:
What about just going with `client=`?  Given typing, it's clear what it needs to be.  This is akin to what many other projects use, where the generic name is clear (by context) to be the namesake- or project-related variant, and other conflicts get the more precise name.

In our project, then, the GlobusCompute client kwarg would be `client=`, and, for example, a Redis client kwarg would be `redis_client=`.

Reply:
IMO changing (funcx_/FUNCX_)* to <blank> is also ok, though client and redis_client are slightly confusing w.r.t. each other.

Reply:
I just pushed up a change to rename it to `client`, but Lei makes a good point. I'll make a thread/poll in dev

Reply:
Poll is up https://funcx.slack.com/archives/C016JMYST9C/p1701799174966749

Reply:
With a landslide 3 to 2 victory, `client` wins 🥇